lib/core/base-provider.js

```diff
@@ -0,0 +1,46 @@
+const providers = require('./providers');
+const path = require('path');
+const ncConfig = require(path.join(process.cwd()), ".nc.config")
+
+class Provider {
+  /**
+   * Provider constructor
+   * @constructor
+   * @param {string} provider - Type of provider - can be found here '/aws/provider'
+   */
+  constructor() {
+    this._providers = this.getProvider();
+    return this._providers;
+  }
+  /**
+   * Returns requested provider type
+   * @param {string} provider - Type of provider - can be found here '/aws/provider'
+   */
+  getProvider() {
+    if (ncConfig === undefined) {
+      throw new Error("Provider config file not found. Please create a config file in project root.");
+    }
+
+    var providersToLoad = {};
+
+    ncConfig.providers.map(provider => {
+      if (provider.name != providers.AWS || provider.name != providers.gcp) {
```

Comment:
This is not a scalable way. In the future, we are planning to implement multiple providers.

Reply:
Should I use the `in` operator?
We will atleast need to add the providers supported in the array for `in` operator to work.
What you think ?